packages/brex/src/index.ts

```diff
@@ -0,0 +1,216 @@
+  idempotencyKey: S.string,
+})
+type CreateTransferArgs = S.To<typeof CreateTransferArgs>
+const createTransfer = (
+  args: CreateTransferArgs
+): Effect.Effect<HTTP.Client, unknown, CreateTransferResponse> =>
+  pipe(
+    Effect.flatMap(BrexHTTPClient, (client) =>
+      pipe(
+        Effect.logDebug('Creating Brex transfer'),
+        Effect.flatMap(() =>
+          client.post({
+            path: '/v1/transfers',
+            headers: {
+              'Content-Type': 'application/json',
+              'Idempotency-Key': args.idempotencyKey,
+            },
+            body: JSON.stringify(args.input),
+          })
+        )
+      )
+    ),
+    Effect.flatMap((response) => Effect.tryPromise(() => response.json())),
+    Effect.tapErrorCause((e) =>
+      Effect.logError('Failed to create Brex transfer', e)
+    )
+  )
+
+const getTransfer = (
+  id: string
+): Effect.Effect<HTTP.Client, unknown, GetTransferResponse> =>
+  Effect.flatMap(BrexHTTPClient, (client) =>
+    pipe(
+      client.get({
+        path: `/v1/transfers?id=${id}`,
+      }),
+      Effect.flatMap((response) => Effect.tryPromise(() => response.json())),
+      Effect.tapErrorCause((e) =>
+        Effect.logError(`Failed to retrieve Brex transfer`, e)
+      )
+    )
+  )
+
+const listTransfers = (
+  cursor?: string,
+  limit?: number
+): Effect.Effect<HTTP.Client, unknown, ListTransfersResponse> =>
+  Effect.flatMap(BrexHTTPClient, (client) =>
+    pipe(
+      Effect.Do,
+      Effect.let(
+        'queryParams',
+        () =>
+          new URLSearchParams(
+            removeUndefined({
+              cursor,
+              limit,
+            })
+          )
+      ),
+      Effect.let('path', ({ queryParams }) => '/v1/transfers?' + queryParams),
+      Effect.bind('response', ({ path }) => client.get({ path })),
+      Effect.flatMap(({ response }) =>
+        Effect.tryPromise(() => response.json())
+      ),
+      Effect.tapErrorCause((e) =>
+        Effect.logError(`Failed to list Brex transfers`, e)
+      )
+    )
+  )
+
+/**
+ * Vendors
+ *
+ * https://developer.brex.com/openapi/payments_api/#tag/Vendors
+ */
+const getVendor = (
+  id: string
+): Effect.Effect<HTTP.Client, unknown, GetVendorResponse> =>
+  pipe(
+    Effect.flatMap(BrexHTTPClient, (client) =>
+      client.get({ path: `/v1/vendors?id=${id}` })
+    ),
+    Effect.flatMap((response) => Effect.tryPromise(() => response.json())),
+    Effect.tapErrorCause((e) =>
+      Effect.logError(`Failed to retrieve Brex vendor`, e)
+    )
+  )
+
+const listVendors = (): Effect.Effect<
+  HTTP.Client,
+  unknown,
+  ListVendorsResponse
+> =>
+  pipe(
+    BrexHTTPClient,
+    Effect.tap(() => Effect.logDebug('Listing Brex vendors')),
+    Effect.flatMap((client) => client.get({ path: `/v1/vendors` })),
+    Effect.flatMap((response) => Effect.tryPromise(() => response.json())),
+    Effect.mapError(
+      (e) =>
+        // @ts-expect-error
+        console.error(e) ||
```

Comment:
I think this was me....:) ok to remove?

Reply:
also, is there still an issue with the http client mangling errors? wondering why I thought this was needed. In general I think we agreed should return as much info as possible from these fns